src/cript/api/utils/save_helper.py

```diff
@@ -1,13 +1,19 @@
-def fix_node_save(api, node, response, known_uuid):
+import re
+from typing import Dict, Optional, Set
+
+
+def _fix_node_save(api, node, response, known_uuid):
     """
     Helper function, that attempts to fix a bad node.
     And if it is fixable, we resave the entire node.
 
     Returns set of known uuids, if fixable, otherwise False.
     """
-    assert response["code"] in (400, 409)
+    if response["code"] not in (400, 409):
+        raise NotImplementedError(f"The internal helper function `_fix_node_save` has been called for an error that is not yet implemented to be handled {response}.")
```

Comment:
I would just throw a regular exception here because it is likely that this is temporary, that way we don't have to create or think about another exception anywhere reducing our code and making our lives easier

```suggestion
    if response["code"] not in (400, 409):
        raise Exception(f"The internal helper function `_fix_node_save` has been called for an error that is not yet implemented to be handled {response}.")
```

Reply:
`Exception` seems an odd choice though.

`RuntimeError` I can see, and I was debating to use, but `Exception` shouldn't really be used on its own.
(Except for blanket catching of Exceptions.)

Reply:
yeah `RuntimeError` probably makes more sense. I usually tend to go with `Exception` whenever I don't have an already created Error and just need a throw away error